### ISSUE TITLE
"No schema available" shown on GraphQL IDE settings page

### ISSUE THREAD

**Issue by benknight** [wp-graphql/wp-graphql#2234]
When opening the GraphQL IDE settings page, the schema fails to load.

There is the following error in the console:

```
devAssert.mjs:5 Uncaught (in promise) Error: Invalid or incomplete introspection result. Ensure that you are passing "data" property of introspection response and no "errors" was returned alongside: undefined.
    at r (devAssert.mjs:5:11)
    at m (buildClientSchema.mjs:27:74)
    at App.js:181:39
```

Upon closer inspection, there is a 500 internal server error when making an async request for the schema:

```json
{
    "errors": [
        {
            "debugMessage": "Return value of GraphQL\\Type\\Schema::resolveType() must be an instance of GraphQL\\Type\\Definition\\Type, null returned",
            "message": "Internal server error",
            "extensions": {
                "category": "internal"
            },
            "locations": [
                {
                    "line": 8,
                    "column": 9
                }
            ],
            "path": [
                "__schema",
                "types"
            ],
            "trace": [
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Type/Definition/FieldDefinition.php",
                    "line": 208,
                    "call": "GraphQL\\Type\\Schema::resolveType(instance of Closure)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Utils/TypeInfo.php",
                    "line": 213,
                    "call": "GraphQL\\Type\\Definition\\FieldDefinition::getType()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Utils/TypeInfo.php",
                    "line": 222,
                    "call": "GraphQL\\Utils\\TypeInfo::extractTypes(GraphQLType: Settings, array(2))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Type/Schema.php",
                    "line": 222,
                    "call": "GraphQL\\Utils\\TypeInfo::extractTypes(GraphQLType: RootQuery, array(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Type/Schema.php",
                    "line": 208,
                    "call": "GraphQL\\Type\\Schema::collectAllTypes()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Type/Introspection.php",
                    "line": 240,
                    "call": "GraphQL\\Type\\Schema::getTypeMap()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 623,
                    "call": "GraphQL\\Type\\Introspection::GraphQL\\Type\\{closure}(instance of WPGraphQL\\WPSchema, array(0), instance of WPGraphQL\\AppContext, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 550,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::resolveFieldValueOrError(instance of GraphQL\\Type\\Definition\\FieldDefinition, instance of GraphQL\\Language\\AST\\FieldNode, instance of Closure, instance of WPGraphQL\\WPSchema, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 1195,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::resolveField(GraphQLType: __Schema, instance of WPGraphQL\\WPSchema, instance of ArrayObject(1), array(2))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 1145,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::executeFields(GraphQLType: __Schema, instance of WPGraphQL\\WPSchema, array(1), instance of ArrayObject(5))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 1106,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::collectAndExecuteSubfields(GraphQLType: __Schema, instance of ArrayObject(1), array(1), instance of WPGraphQL\\WPSchema)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 793,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::completeObjectValue(GraphQLType: __Schema, instance of ArrayObject(1), instance of GraphQL\\Type\\Definition\\ResolveInfo, array(1), instance of WPGraphQL\\WPSchema)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 741,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::completeValue(GraphQLType: __Schema, instance of ArrayObject(1), instance of GraphQL\\Type\\Definition\\ResolveInfo, array(1), instance of WPGraphQL\\WPSchema)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 654,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::completeValue(GraphQLType: __Schema, instance of ArrayObject(1), instance of GraphQL\\Type\\Definition\\ResolveInfo, array(1), instance of WPGraphQL\\WPSchema)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 557,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::completeValueCatchingError(GraphQLType: __Schema, instance of ArrayObject(1), instance of GraphQL\\Type\\Definition\\ResolveInfo, array(1), instance of WPGraphQL\\WPSchema)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 1195,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::resolveField(GraphQLType: RootQuery, null, instance of ArrayObject(1), array(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 264,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::executeFields(GraphQLType: RootQuery, null, array(0), instance of ArrayObject(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
                    "line": 215,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::executeOperation(instance of GraphQL\\Language\\AST\\OperationDefinitionNode, null)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Executor/Executor.php",
                    "line": 156,
                    "call": "GraphQL\\Executor\\ReferenceExecutor::doExecute()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/GraphQL.php",
                    "line": 162,
                    "call": "GraphQL\\Executor\\Executor::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of WPGraphQL\\WPSchema, instance of GraphQL\\Language\\AST\\DocumentNode, null, instance of WPGraphQL\\AppContext, null, null, null)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Server/Helper.php",
                    "line": 311,
                    "call": "GraphQL\\GraphQL::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of WPGraphQL\\WPSchema, instance of GraphQL\\Language\\AST\\DocumentNode, null, instance of WPGraphQL\\AppContext, null, null, null, array(29))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Server/Helper.php",
                    "line": 211,
                    "call": "GraphQL\\Server\\Helper::promiseToExecuteOperation(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of GraphQL\\Server\\ServerConfig, instance of GraphQL\\Server\\OperationParams)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/vendor/webonyx/graphql-php/src/Server/StandardServer.php",
                    "line": 136,
                    "call": "GraphQL\\Server\\Helper::executeOperation(instance of GraphQL\\Server\\ServerConfig, instance of GraphQL\\Server\\OperationParams)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/src/Request.php",
                    "line": 679,
                    "call": "GraphQL\\Server\\StandardServer::executeRequest(instance of GraphQL\\Server\\OperationParams)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/src/Router.php",
                    "line": 463,
                    "call": "WPGraphQL\\Request::execute_http()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-content/plugins/wp-graphql/src/Router.php",
                    "line": 265,
                    "call": "WPGraphQL\\Router::process_http_request()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/class-wp-hook.php",
                    "line": 307,
                    "call": "WPGraphQL\\Router::resolve_http_request(instance of WP)"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/class-wp-hook.php",
                    "line": 331,
                    "call": "WP_Hook::apply_filters(null, array(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/plugin.php",
                    "line": 522,
                    "call": "WP_Hook::do_action(array(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/class-wp.php",
                    "line": 396,
                    "function": "do_action_ref_array('parse_request', array(1))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/class-wp.php",
                    "line": 758,
                    "call": "WP::parse_request((empty string))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-includes/functions.php",
                    "line": 1310,
                    "call": "WP::main((empty string))"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/wp-blog-header.php",
                    "line": 16,
                    "function": "wp()"
                },
                {
                    "file": "/home/customer/www/cms.vietnamcoracle.com/public_html/index.php",
                    "line": 17,
                    "function": "require('/home/customer/www/cms.vietnamcoracle.com/public_html/wp-blog-header.php')"
                }
            ]
        }
    ],
    "extensions": {
        "debug": []
    }
}
```

This issue has actually been persisting for months now.

WordPress 5.9
WPGraphQL 1.6.12

**Comment by benknight** [comment]
Update, I've tried deactivating and reactivating all additional plugins that interact with WP GraphQL in any way, and still received the error in all cases.

**Comment by jacraven** [comment]
Having the same issue.

**Comment by sinisteralex** [comment]
I have the same issue.

**Comment by jasonbahl** [comment]
🤔  can ya'll provide more information about your environment, other active themes and plugins, PHP versions, etc?

I use GraphiQL every day and have not ran into this issue on any version of WPGraphQL.

**Comment by jasonbahl** [comment]
@sinisteralex @jacraven @benknight if possible, please provide steps to reproduce. 

I use local by flywheel to spin up WordPress environments quickly. 

If I start a fresh WordPress install in Local by Flywheel, then activate WPGraphQL, I can use WPGraphQL and the GraphiQL IDE in the dashboard, without issue. 

Here's a screenshot of my local setup:

![CleanShot 2022-02-25 at 15 41 56](https://user-images.githubusercontent.com/1260765/155812873-fb3ebafd-43e6-497a-a819-30699283759c.png)

PHP 8.0
MySQL 8.0.16
WordPress 5.9.1

### Screenshot of GraphiQL (with Schema loaded) in v1.6.12

![CleanShot 2022-02-25 at 15 44 17](https://user-images.githubusercontent.com/1260765/155813059-f6b3f368-66cc-4439-8019-9e234b603642.png)


Screenshot of GraphiQL (with Schema loaded) in the upcoming v1.7 release

![CleanShot 2022-02-25 at 15 43 11](https://user-images.githubusercontent.com/1260765/155812971-f5056da8-8756-4600-bb65-97d75549de79.png)

**Comment by jasonbahl** [comment]
👋🏻  there's an issue related to settings that are registered with the core WordPress `register_setting()` API that are set to `show_in_rest` but don't map to a proper GraphQL Type. 

Do you happen to have plugins or theme code that makes use of the `register_setting()` function, and sets the setting(s) to `show_in_rest => true`? If so, this issue is likely related: https://github.com/wp-graphql/wp-graphql/issues/2065

If not, I'll still need steps to reproduce. What theme to activate, etc, etc.

**Comment by sinisteralex** [comment]
My Local by flywheel setup

![image](https://user-images.githubusercontent.com/53612147/155846465-85bcbf4f-9435-4ad4-84d1-ebec1015c768.png)

My Wordpress GraphiQL IDE

![image](https://user-images.githubusercontent.com/53612147/155846486-bf37a8d0-f11e-4eee-94bd-598a150364d7.png)
My GraphQL versions 

![image](https://user-images.githubusercontent.com/53612147/155846552-cdd209f9-f0a1-4253-b25e-01bf2762e768.png)

**Comment by benknight** [comment]
Hey @jasonbahl I sent you an email about this that should hopefully help debug this issue.

**Comment by jasonbahl** [comment]
@benknight thanks! Got the email. I think the conflict _might_ be the Stackable plugin. 

I have a lead on fixing this. It does seem related to the `register_settings()` bug that I mentioned above. I'll try and get a PR together this week to address this situation. 

(I'll also see if I can get to the bottom of the search pagination thing we've talked about a few times!)

**Comment by jasonbahl** [comment]
@sinisteralex what _other_ plugins do you have active? My hunch is that it's a bug with code not directly related to WPGraphQL or WPGraphQL extensions. I believe that any theme/plugin that makes use of the WordPress core `register_setting` function might be the conflict. I'd like to be able to confirm that, if possible. If that's not the case, then we might have another bug that will require further troubleshooting to identify.

**Comment by sinisteralex** [comment]
![image](https://user-images.githubusercontent.com/53612147/155969669-b2def4c1-a54b-419c-ab6d-5e5f1e7f8676.png)

**Comment by jasonbahl** [comment]
Awesome! Thanks for confirming. 🙏 

I hope to have this patched this week

**Comment by jasonbahl** [comment]
@benknight @sinisteralex @jacraven I have a PR open here: #2262 that I believe fixes this issue. 

I've tested with 2 plugins that are known to cause the conflict: 

- Google Web Stories
- Stackable (Gutenberg blocks)

With those plugins active, along with the changes in this PR (https://github.com/wp-graphql/wp-graphql/pull/2262), the Schema loads in the wp-admin without issue, so you can use GraphiQL again. 

If you have time, please check that the changes applied in this PR (https://github.com/wp-graphql/wp-graphql/pull/2262) allow you to use GraphiQL again with Stackable and/or Google Web Stories active. 

If there's still an issue after testing with these changes, I'll need more information to try and reproduce the situation. 

Thanks! 🙏🏻

**Comment by jasonbahl** [comment]
I'm currently running both Stackable and Google Web Stories with this PR and WPGraphQL is working. I'm going to consider this resolved per #2262 which is part of the v1.7.0 release. 

If, after updating to v1.7.0 you feel this issue is still happening, please open a new issue, and feel free to reference this one for context. 

Thanks! 🙏🏻
